### PR TITLE
[#2957] Upgrade to jshint 0.9.1 and fixed all lint errors from doing so

### PR DESCRIPTION
**Pull request by mjschranz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -83,7 +83,7 @@
   },
   "dependencies": {
     "shelljs": "~0.0.9",
-    "jshint": "0.6.2",
+    "jshint": "0.9.1",
```
Comment (jbuck) [inline package.json]: Use the "~0.9.1" syntax here. You'll also need to regenerate the shrinkwrap file using `npm shrinkwrap` once you've installed v0.9.1.

File: src/core/project.js [hunk 2]
```diff
@@ -2,8 +2,8 @@
  * If a copy of the MIT license was not distributed with this file, you can
  * obtain one at https://raw.github.com/mozilla/butter/master/LICENSE */
 
-define( [ "core/eventmanager", "core/media", "util/shims" ],
```
Comment (jbuck) [inline src/core/project.js:5]: I think you'll need to leave "util/shims" in the list here; just loading it adds the shims properly. You don't need the Shims var though.

File: src/cornfield/module.js [hunk 3]
```diff
@@ -2,8 +2,8 @@
  * If a copy of the MIT license was not distributed with this file, you can
  * obtain one at https://raw.github.com/mozilla/butter/master/LICENSE */
 
-define([ "util/xhr", "jsSHA/sha1", "util/shims" ],
```
Comment (jbuck) [inline src/cornfield/module.js:5]: jsSHA/sha1 needs to be left there, PM crashes otherwise

File: src/first-run.js [hunk 4]
```diff
@@ -7,7 +7,7 @@
  *
  * Determines whether or not a user should be shown a first-run dialog
  */
-define( [ "dialog/dialog", "ui/widget/tooltip", "util/shims" ], function( Dialog, ToolTip, Shims ) {
```
Comment (jbuck) [inline src/first-run.js:10]: Restore "util/shims"

File: templates/assets/editors/googlemap/googlemap-editor.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/*global google*/
+/*global */
```
Comment (jbuck) [inline templates/assets/editors/googlemap/googlemap-editor.js]: You can just delete this comment entirely if the "google" global isn't being used

File: src/cornfield/module.js (hunk 3 above)
Comment (secretrobotron) [inline src/cornfield/module.js:5]: Add a space after the first round brace here.

**Pull request merged**